### PR TITLE
main.py, portage.util: Add docstrings and type annotations

### PR DESCRIPTION
**Pull request by berinaniesh**
added a few more items to `.gitignore` also.

### CODE REVIEW

**Comment by berinaniesh** [comment]
I ran black on the modified files before committing, but the CI fails here. I am guessing it is because of the python 3.9+ style type annotations. What should I do @thesamesam ?

**Review by thesamesam — CHANGES_REQUESTED** [review]

File: lib/portage/util/__init__.py
Comment (thesamesam) [inline lib/portage/util/__init__.py:133]: Shouldn't we annotate this function while we're here? It's very similar to the other.

Comment (berinaniesh) [reply lib/portage/util/__init__.py:133]: Did you mean to add return type to `writemsg_level`? My bad, just did. Pushing it.

**Comment by thesamesam** [comment]
lgtm, will merge when CI is done, thanks!

**Comment by thesamesam** [comment]
Missing signoff, please `git rebase --signoff -f` and push

**Comment by berinaniesh** [comment]
> Missing signoff, please `git rebase --signoff -f` and push

But I have signoffs in all three commits. Git push says everything up to date.

**Comment by thesamesam** [comment]
Ah, you manually wrote the signoff: it's `Signed-off-by`, not `Signed-Off-by`, so some tools don't recognise it, like pram:
```
2023-06-06 07:47:06 (119 MB/s) - ‘/tmp/tmp.jqur4VRL4a/all.patch’ saved [7316]

Commit no. 0001 was not signed off by the author!
```

**Comment by berinaniesh** [comment]
> Ah, you manually wrote the signoff: it's `Signed-off-by`, not `Signed-Off-by`, so some tools don't recognise it, like pram:
> 
> ```
> 2023-06-06 07:47:06 (119 MB/s) - ‘/tmp/tmp.jqur4VRL4a/all.patch’ saved [7316]
> 
> Commit no. 0001 was not signed off by the author!
> ```

I did. I had git do it automatically in the local repo because I don't want to sign off other commits I do. I often delete the source repo and resync. It resets. So, I manually add Sign offs. Next time I'll take care of the case. Is that fine?

**Comment by thesamesam** [comment]
>I often delete the source repo and resync.

Please don't do this, it's an antipattern. You should never need to. If you think you're in a situation where you do, ask for help.

**Comment by thesamesam** [comment]
Anyway, yes, fine, I'll fix it here, although I think you could've with `git rebase -i` quickly?

**Comment by ajakk** [comment]
Thanks for working on this!

**Pull request closed without merging**
